allennlp/common/checks.py

```diff
@@ -33,3 +36,16 @@ def check_dimensions_match(dimension_1: int,
     if dimension_1 != dimension_2:
         raise ConfigurationError(f"{dim_1_name} must match {dim_2_name}, but got {dimension_1} "
                                  f"and {dimension_2} instead")
+
+
+def cuda_is_valid(device_id: int):
+    if device_id is None:
+        # this branch is to ensure that existing tests which don't specify cuda pass.
+        return
+    try:
+        with cuda.device(device_id):
+            pass
+    except Exception:
+        raise ConfigurationError("Experiment specified a GPU but none is available;"
```

Comment:
This `if / return / try / with / pass / except` construction feels way too complicated for what it does.

Why not something like just

```
if device_id is not None and device_id >= cuda.device_count():
    raise ConfigurationError(...)
```

Would that accomplish the same?